mods/ra/maps/allies-11a/allies11a.lua

```diff
@@ -0,0 +1,203 @@
+				a.Patrol(sea1PatrolPath, false, DateTime.Seconds(2))
+			end)
+			Trigger.OnEnteredFootprint(edgeOfRiverTriggerActivator, function(a)
+				if a.Owner == England and a.Type == "dd" or a.Type == "pt" then
+					a.Destroy()
+				end
+			end)
+		end)
+		local sea2Units = Reinforcements.Reinforce(England, sea2Reinforcements.actors, sea2Reinforcements.entryPath)
+		Utils.Do(sea2Units, function(a)
+			Trigger.OnAddedToWorld(a, function()
+				a.Patrol(sea2PatrolPath, false, DateTime.Seconds(2))
+			end)
+			Trigger.OnEnteredFootprint(edgeOfRiverTriggerActivator, function(a)
+				if a.Owner == England and a.Type == "dd" or a.Type == "pt" then
+					a.Destroy()
+				end
+			end)
+		end)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(10), timerExpiredSendCa)
+end
+
+timerExpiredSendCa = function()
+	Trigger.AfterDelay(DateTime.Seconds(30), function()
+		local sea3Units = Reinforcements.Reinforce(England, sea3Reinforcements.actors, sea3Reinforcements.entryPath)
+		Utils.Do(sea3Units, function(a)
+			Trigger.OnAddedToWorld(a, function()
+				a.Move(WP57.Location, 0)
+			end)
+			Trigger.OnKilled(a, function()
+				USSR.MarkCompletedObjective(USSRObj)
+			end)
+		end)
+		local sea4Units = Reinforcements.Reinforce(England, sea4Reinforcements.actors, sea4Reinforcements.entryPath)
+		Utils.Do(sea4Units, function(a)
+			Trigger.OnAddedToWorld(a, function()
+				a.Move(WP65.Location, 0)
+			end)
+			Trigger.OnKilled(a, function()
+				USSR.MarkCompletedObjective(USSRObj)
+			end)
+		end)
+		count = 0
+		Trigger.OnEnteredFootprint(edgeOfRiverTriggerActivator, function(a, id)
+			if a.Owner == England and a.Type =="ca" then
+				count = count + 1
+				a.Destroy()
+				if count == 2 then
+					Greece.MarkCompletedObjective(ClearNavalChannel)
+					Trigger.RemoveFootprintTrigger(id)
+				end		
+			end			
+		end)
+	end)
+end
+
+ticked = TimerTicks
+Tick = function()
+	USSR.Cash = 5000
+	BadGuy.Cash = 10000
+	if Greece.HasNoRequiredUnits() then
+		USSR.MarkCompletedObjective(USSRObj)
+	end
+	if ticked > 0 then
+		UserInterface.SetMissionText("Naval vessels arrive in " .. Utils.FormatTime(ticked), TimerColor)
+		ticked = ticked - 1
+		if USSR.HasNoRequiredUnits() and BadGuy.HasNoRequiredUnits() then
+			ticked = 0
+		end
+	elseif ticked == 0 then
+		FinishTimer()
+		timerExpiredSendNavy()
+		ticked = ticked - 1
+	end
+end
+
+FinishTimer = function()
+	for i = 0, 9, 1 do
+		local c = TimerColor
+		if i % 2 == 0 then
+			c = HSLColor.White
+		end
+		Trigger.AfterDelay(DateTime.Seconds(i), function() UserInterface.SetMissionText("Naval vessels have arrived!", c) end)
+	end
+	Trigger.AfterDelay(DateTime.Seconds(10), function() UserInterface.SetMissionText("") end)
+end
+
+WorldLoaded = function()
+	Greece = Player.GetPlayer("Greece")
+	USSR = Player.GetPlayer("USSR")
+	BadGuy = Player.GetPlayer("BadGuy")
+	England = Player.GetPlayer("England")
+	Camera.Position = WP98.CenterPosition
+	ClearNavalChannel = Greece.AddPrimaryObjective("Clear the naval channel.")
```

Comment:
Use `AddObjective` here. We're phasing out `AddPrimaryObjective` and `AddSecondaryObjective`.

Reply:
Change AddPrimaryObjective to AddObjective